### PR TITLE
Validate JSON snippets are not only expected, but valid.

### PR DESCRIPTION
**Pull request by billf**

### CODE REVIEW

File: field_test.go [hunk 1]
```diff
@@ -46,9 +47,19 @@ func assertFieldJSON(t testing.TB, expected string, field Field) {
 	enc := newJSONEncoder()
 	defer enc.Free()
 
+	var out interface{}
+	err := json.Unmarshal([]byte("{"+expected+"}"), &out)
+	require.NoError(t, err,
+		"Expected JSON snippet '%s' must be valid for use in an object.", expected)
```
Comment (prashantv) [inline field_test.go]: nit: use `%q` rather than `'%s'`

File: field_test.go [hunk 2]
```diff
@@ -46,9 +47,19 @@ func assertFieldJSON(t testing.TB, expected string, field Field) {
...
+
 	field.AddTo(enc)
 	assert.Equal(t, expected, string(enc.bytes),
 		"Unexpected JSON output after applying field %+v.", field)
+	eb := append([]byte{'{'}, enc.bytes...)
```
Comment (prashantv) [inline field_test.go]: once we validate the expected unmarshals with `{` and `}` around it, and we make sure that `expected` is the same as `enc.bytes`, is there any point in trying to unmarshal the `enc.bytes` separately?

Comment (billf) [inline field_test.go]: none that i can think of. 

i did this section first and then thought "i should also validate that what's passed in as expected is valid as well"

**Comment by prashantv** [comment]
lgtm, thanks @billf

**Pull request merged**